test/common/schemaform/fields/ArrayField.unit.spec.jsx

```diff
@@ -239,4 +241,181 @@ describe('Schemaform <ArrayField>', () => {
 
     expect(tree.subTree('button').props.disabled).to.be.true;
   });
+  it('should render review mode when specified and has data', () => {
+    const idSchema = {};
+    const schema = {
+      type: 'array',
+      items: [],
+      additionalItems: {
+        type: 'object',
+        properties: {
+          field: {
+            type: 'string'
+          }
+        }
+      }
+    };
+    const formData = [
+      {},
+    ];
+    const uiSchema = {
+      'ui:title': 'List of things',
+      'ui:options': {
+        showLastItemInViewMode: true,
+        viewField: f => f
+      }
+    };
+    const fullErrorSchema = {
+      0: {},
+    };
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const tree = SkinDeep.shallowRender(
+      <ArrayField
+        schema={schema}
+        errorSchema={fullErrorSchema}
+        uiSchema={uiSchema}
+        idSchema={idSchema}
+        registry={registry}
+        onChange={onChange}
+        onBlur={onBlur}
+        formData={formData}
+        formContext={formContext}
+        requiredSchema={requiredSchema}/>
+    );
+    expect(tree.everySubTree('SchemaField').length).to.equal(0);
+  });
+  it('should render edit mode when specified and has no data', () => {
+    const idSchema = {};
+    const schema = {
+      type: 'array',
+      items: [],
+      additionalItems: {
+        type: 'object',
+        properties: {
+          field: {
+            type: 'string'
+          }
+        }
+      }
+    };
+    const formData = [
+    ];
+    const uiSchema = {
+      'ui:title': 'List of things',
+      'ui:options': {
+        showLastItemInViewMode: true,
+        viewField: f => f
+      }
+    };
+    const fullErrorSchema = {
+    };
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const tree = SkinDeep.shallowRender(
+      <ArrayField
+        schema={schema}
+        errorSchema={fullErrorSchema}
+        uiSchema={uiSchema}
+        idSchema={idSchema}
+        registry={registry}
+        onChange={onChange}
+        onBlur={onBlur}
+        formData={formData}
+        formContext={formContext}
+        requiredSchema={requiredSchema}/>
+    );
+    expect(tree.everySubTree('SchemaField').length).to.equal(1);
+  });
+  it('should render edit mode when specified and has invalid data', () => {
```

Comment:
Same as above but with `invalid data is given`.

Reply:
True, will update similarly.